packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -94,39 +91,33 @@ function CustomSelectControl( props: LegacyCustomSelectProps ) {
 		);
 	};
 
-	// translate legacy button sizing
-	const contextSystemValue = useMemo( () => {
-		let selectedSize;
-
+	const translatedSize = ( () => {
```

Comment:
Is there a good reason for the IIFE here? Can be a simple `if` without an IIFE IMHO.

Reply:
Touché, I should've said _reassignment_! I always considered variable reassignment to be a form of mutation (as in mutable state), but I guess those two terms should be differentiated in JS at least.